### PR TITLE
feat(Pattern): add new Patterns and rename some old ones

### PR DESCRIPTION
**Pull request by natalieweesh**
### Overview

add some new patterns and rename dots to checker and rename diagonalStripes to diagonalStripeB

### PR Checklist

- [ ] Related to designs: https://www.figma.com/file/j0vUEMpEJyA5lf5J0uPgJC/Pattern-Fills---12.14.20?node-id=0%3A1
- [ ] Related to JIRA ticket: EGG-1085
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

Screen Shot 2021-04-19 at 2 25 09 PM

Screen Shot 2021-04-19 at 2 25 15 PM

Screen Shot 2021-04-19 at 2 25 21 PM

Screen Shot 2021-04-19 at 2 25 39 PM

### CODE REVIEW

File: packages/gamut/src/Pattern/index.tsx [hunk 1]
```diff
@@ -5,12 +5,36 @@ import { Box, BoxProps } from '../Box';
...
+  | 'gridDense'
+  | 'herringboneLoose'
+  | 'herringboneRegular'
+  | 'herringboneDense'
+  | 'rainLoose'
+  | 'rainRegular'
+  | 'rainDense'
+  | 'stripeLoose'
+  | 'stripeRegular'
+  | 'stripeDense';
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Pattern/index.tsx:37]: lol why do we have so many of these...

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
If we _do_ have dozens of different patterns, let's split them up into separate components the way icons and illustrations are. We're getting to the point where that'd be both more scalable and better for performance _(such a huge file to always import!)_

File: packages/gamut/src/Pattern/index.tsx (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/Pattern/index.tsx:37]: asking on Slack: https://codecademy.slack.com/archives/CTVEHH2PP/p1618862487000400

**Comment by natalieweesh** [comment]
closing in favor of #1654 which creates a new package for patterns instead

**Pull request closed without merging**